src/nuclide_header.F90

```diff
@@ -0,0 +1,761 @@
+module nuclide_header
+
+  use, intrinsic :: ISO_FORTRAN_ENV
+
+  use ace_header
+  use constants
+  use endf,        only: reaction_name
+  use list_header, only: ListInt
+  use math,        only: evaluate_legendre
+  use string
+
+  implicit none
+
+!===============================================================================
+! NuclideBase contains the base nuclidic data for a nuclide, which does not depend
+! upon how the nuclear data is represented (i.e., CE, or any variant of MG).
+! The extended types, NuclideCE and NuclideMG deal with the rest
+!===============================================================================
+
+  type, abstract :: NuclideBase
```

Comment:
I would rename this to just `Nuclide`. None of our other abstract types use a "Base" suffix.